index.js

```diff
@@ -20,23 +20,116 @@
   }
 })();
 
-const daysList = [
-  "",
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-  "Sunday",
-];
-
-for (let i = 0; i < daysList.length; i++) {
-  let day = daysList[i];
-  const dayTextNode = document.createTextNode(day);
-  const Div = document.createElement("div");
-  Div.setAttribute("class", "day");
-  Div.appendChild(dayTextNode);
-  document.querySelector(".day-container").appendChild(Div);
+(function () {
+  const daysList = [
+    "",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+    "Sunday",
+  ];
+
+  for (let i = 0; i < daysList.length; i++) {
+    let day = daysList[i];
+    const dayTextNode = document.createTextNode(day);
+    const Div = document.createElement("div");
+    Div.setAttribute("class", "day");
+    Div.appendChild(dayTextNode);
+    document.querySelector(".day-container").appendChild(Div);
+  }
+  document.querySelector(".day").setAttribute("id", "corner");
+})();
+
+(function () {
+  const subjectsField = [
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+    "",
+  ];
+
+  for (let i = 0; i < subjectsField.length; i++) {
+    let subjectField = subjectsField[i];
+    const subjectFieldTextNode = document.createTextNode(subjectField);
+    const div = document.createElement("div");
+    div.setAttribute("class", "subject");
+    div.setAttribute("ondrop", "drop(event)");
+    div.setAttribute("ondragover", "allowDrop(event)");
+    div.appendChild(subjectFieldTextNode);
+    document.querySelector(".dragField").appendChild(div);
+  }
+
+  for (let i = 0; i < subjectsField.length; i++) {
+    let subjectField = subjectsField[i];
+    const subjectFieldTextNode = document.createTextNode(subjectField);
+    const div = document.createElement("div");
+    div.setAttribute("class", "subject");
+    div.setAttribute("ondrop", "drop(event)");
+    div.setAttribute("ondragover", "allowDrop(event)");
+    div.appendChild(subjectFieldTextNode);
+    document.getElementsByClassName("dragField")[1].appendChild(div);
+  }
+})();
+
+function allowDrop(event) {
+  event.preventDefault();
+}
+
+function drag(event) {
+  event.dataTransfer.setData("text", event.target.id);
+}
+
+function drop(event) {
+  event.preventDefault();
+  var data = event.dataTransfer.getData("text");
+  event.target.appendChild(document.getElementById(data));
+}
+
+document.getElementsByClassName("subject__name")[0].setAttribute("id", "math");
```

Comment:
hardcode is now allowed.
Reconstruct `subjectList` as an array of objects which contains `id`, `name` and `backgroundColor`